penguin/integrations/mcp/manager.py

```
@@ -0,0 +1,497 @@
+            params = StdioServerParameters(  # type: ignore[misc]
+                command=state.config.command,
+                args=state.config.args,
+                env=state.config.env or None,
+                cwd=state.config.cwd,
+            )
+            read_stream, write_stream = await stack.enter_async_context(
+                stdio_client(params)
+            )
+        elif state.config.transport == "streamable_http":
+            if streamablehttp_client is None:
+                raise RuntimeError("MCP streamable HTTP client is unavailable")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)